pkg/ccl/crosscluster/streamclient/randclient/random_stream_client.go

```diff
@@ -232,23 +248,31 @@ type randomEventGenerator struct {
 	rng                        *rand.Rand
 	config                     randomStreamConfig
 	numEventsSinceLastResolved int
-	sstMaker                   SSTableMakerFn
+	sstMaker                   streamclient.SSTableMakerFn
 	codec                      keys.SQLCodec
-	tableDesc                  *tabledesc.Mutable
+	tableDesc                  catalog.TableDescriptor
 	systemKVs                  []roachpb.KeyValue
 }
 
 func newRandomEventGenerator(
-	rng *rand.Rand, partitionURL *url.URL, config randomStreamConfig, fn SSTableMakerFn,
+	rng *rand.Rand,
+	partitionURL *url.URL,
+	config randomStreamConfig,
+	fn streamclient.SSTableMakerFn,
+	tableDesc catalog.TableDescriptor,
 ) (*randomEventGenerator, error) {
 	var partitionTableID int
 	partitionTableID, err := strconv.Atoi(partitionURL.Host)
 	if err != nil {
 		return nil, err
 	}
-	tableDesc, systemKVs, err := getDescriptorAndNamespaceKVForTableID(config, descpb.ID(partitionTableID))
-	if err != nil {
-		return nil, err
+
+	var systemKVs []roachpb.KeyValue
```

Comment:
no need to change this now, but it's not clear to me what this systemKVs knob is adding to test coverage.

Reply:
Not clear to me either. Perhaps the original idea was that these would be needed to allow us to read from the table post promotion. But the tests don't actually do that. And, even if they did, all it would be testing is that the table is readable, there would be nothing to compare it to.